### DISCUSSION TITLE
PerformanceTest Pyramid Scene is not deterministic

### DISCUSSION THREAD

**Discussion by RockDDD** [jrouwe/JoltPhysics#1336] [Q&A] [answered]
turn on the CROSS_PLATFORM_DETERMINISTIC in the CMakeLists.txt, i find the Pyramid Scene test is not deterministic. the hash result on windows platform is different from other platforms.

**Comment by RockDDD** [comment]
i compare platforms with macosx (x86-x64), android(arm64-v8a), ios(arm64),windows(vs2022 clang-cl). the result of windows is different to others

**Comment by jrouwe** [comment]
Thanks for the report, I'll investigate!

**Comment by jrouwe** [comment]
FYI, the Ragdoll test is broken now too. An update to a newer MSVC version broke it: 19.41.34123.0 -> 19.42.34433.0.

Are you on that new version?

**Reply by RockDDD** [reply]
my version is 19.36.32537

**Comment by jrouwe** [comment, marked as answer]
The issue should have been fixed with #1351

**Reply by RockDDD** [reply]
setting Discrete and LinearCast causes the ConvexVsMesh test results to differ,is that right?

**Reply by jrouwe** [reply]
Yes that is expected, you change the way collision detection works if you do that.

**Reply by RockDDD** [reply]
ok, thank you very much!
